### PR TITLE
tsdb: Fixed Symbol Lookup edge case; Added tests.

### PR DESCRIPTION
**Pull request by bwplotka**
Without fix lookup for last reference + 1 returns extra bytes after symbols.

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: tsdb/index/index_test.go [hunk 1]
```diff
@@ -505,3 +506,48 @@ func TestNewFileReaderErrorNoOpenFiles(t *testing.T) {
...
 	dir.Close()
 }
+
+func TestSymbols(t *testing.T) {
+	buf := encoding.Encbuf{}
+
+	buf.PutUvarintStr("something")
+
+	symbolsStart := buf.Len()
+	buf.PutBE32int(204)
```
Comment (brian-brazil) [inline tsdb/index/index_test.go]: This could all do with comments so you know what field is what.

File: tsdb/index/index_test.go [hunk 2]
```diff
@@ -505,3 +506,48 @@ func TestNewFileReaderErrorNoOpenFiles(t *testing.T) {
...
+	buf.PutBE32int(100)
+	for i := 0; i < 100; i++ {
+		buf.PutUvarintStr(string(i))
```
Comment (brian-brazil) [inline tsdb/index/index_test.go]: These symbols will not be in order, you'd need fixed size strings.

Comment (brian-brazil) [reply tsdb/index/index_test.go]: Or add your test to TestPostingsMany, which already has all the required infrastructure.

Comment (bwplotka) [reply tsdb/index/index_test.go]: I think a separate unit test for separate symbols structure would be nice (:

Additionally, those symbols are in perfect order, when you print `string(i)`:   `       	 
                     ! " # $ % & ' ( ) * + , - . / 0 1 2 3 4 5 6 7 8 9 : ; < = > ? @ A B C D E F G H I J K L M N O P Q R S T U V W X Y Z [ \ ] ^ _ ` a b c `

They would be not in order if I would do `fmt.Sprint(i)` (:

File: tsdb/index/index_test.go (hunk 1 above)
Comment (bwplotka) [reply tsdb/index/index_test.go]: :+1:

File: tsdb/index/index_test.go (hunk 2 above)
Comment (brian-brazil) [reply tsdb/index/index_test.go]: That needs a comment then, I'd intuitively expect it to be the decimal representation of the int

Comment (bwplotka) [reply tsdb/index/index_test.go]: done

File: tsdb/index/index_test.go [hunk 3]
```diff
@@ -505,3 +506,50 @@ func TestNewFileReaderErrorNoOpenFiles(t *testing.T) {
...
+
+	symbolsStart := buf.Len()
+	buf.PutBE32int(204) // Length of symbols table.
+	buf.PutBE32int(100) // Number of symbols.
+	for i := 0; i < 100; i++ {
+		// i represents index in unicode characters table.
+		buf.PutUvarintStr(string(i)) // Symbol.
+	}
+	hash := crc32.Checksum(buf.Get()[symbolsStart+4:], castagnoliTable)
+	buf.PutBE32(hash) // Check sum at the end.
```
Comment (brian-brazil) [inline tsdb/index/index_test.go]: Checksum

**Pull request merged**